src/riak_kv_vnode.erl

```diff
@@ -1130,41 +1205,78 @@ handle_command(tictacaae_exchangepoke, _Sender, State) ->
-                    aae_exchange:start(BlueList, 
-                                        PinkList, 
-                                        RepairFun, 
-                                        ReplyFun);
+            SkipCount =
+                case {lists:keyfind(Local, 1, PL),
+                        lists:keyfind(Remote, 1, PL)} of
+                    {{Local, LN}, {Remote, RN}} ->
+                        IndexN = {DocIdx, N},
+                        BlueList = 
+                            [{riak_kv_vnode:aae_send({Local, LN}), [IndexN]}],
+                        PinkList = 
+                            [{riak_kv_vnode:aae_send({Remote, RN}), [IndexN]}],
+                        RepairFun = 
+                            riak_kv_get_fsm:prompt_readrepair([{Local, LN}, 
+                                                                {Remote, RN}]),
+                        ReplyFun = tictac_returnfun(Idx, exchange),
+                        ScanTimeout = ?AAE_SKIP_COUNT * XTick,
+                        ExchangeOptions =
+                            case app_helper:get_env(riak_kv,
+                                                    tictacaae_maxresults) of
+                                MR when is_integer(MR) ->
+                                    [{scan_timeout, ScanTimeout},
+                                        {max_results, MR}];
+                                _ ->
+                                    [{scan_timeout, ScanTimeout}]
+                            end,
+                        aae_exchange:start(full,
+                                            BlueList, 
+                                            PinkList, 
+                                            RepairFun, 
+                                            ReplyFun,
+                                            none,
+                                            ExchangeOptions),
+                        ?AAE_SKIP_COUNT;
                 _ ->
                     lager:warning("Proposed exchange between ~w and ~w " ++ 
                                     "not currently supported within " ++
-                                    "preflist for IndexN=~w possibly due to " ++
-                                    "node failure",
-                                    [Local, Remote, {DocIdx, N}])
+                                    "preflist for IndexN=~w possibly " ++
+                                    "due to node failure",
+                                    [Local, Remote, {DocIdx, N}]),
+                        0
             end,
             ok = aae_controller:aae_ping(State#state.aae_controller,
                                             os:timestamp(),
                                             self()),
-            {noreply, State#state{tictac_exchangequeue = Rest}}
+            {noreply, State#state{tictac_exchangequeue = Rest,
+                                    tictac_skiptick = SkipCount}};
+        {_, SkipCount} ->
```

Comment:
Unlikely, but potentially the Skipcount can get below zero.
This would happen when the exchange state is a non-empty list with an element of the wrong format. So, it would skip the case above. Most likely you then have bigger problems then a negative skipcount.

But you could do your max trick here.